blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java

```diff
@@ -235,57 +231,14 @@ public HttpResponse validateAndCreate(@JsonBody JSONObject request) {
             final StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER, "github", "Github Access Token", user.getLogin(), accessToken);
 
 
-            CredentialsStore store=null;
-            for(CredentialsStore s: CredentialsProvider.lookupStores(authenticatedUser)){
-                if(s.hasPermission(CredentialsProvider.CREATE) && s.hasPermission(CredentialsProvider.UPDATE)){
-                    store = s;
-                    break;
-                }
-            }
-
-            if(store == null){
-                throw new ServiceException.ForbiddenException(String.format("Logged in user: %s doesn't have writable credentials store", authenticatedUser.getId()));
-            }
-
-            Domain domain = store.getDomainByName(DOMAIN_NAME);
-            if(domain == null){
-                java.net.URI uri = new URI(getUri());
-
-                List<DomainSpecification> domainSpecifications = new ArrayList<>();
-
-                // XXX: UriRequirementBuilder.fromUri() maps "" path to "/", so need to take care of it here
-                String path = uri.getRawPath() == null ? null : (uri.getRawPath().trim().isEmpty() ? "/" : uri.getRawPath());
-                domainSpecifications.add(new PathSpecification(path, "", false));
-                if(uri.getPort() != -1){
-                    domainSpecifications.add(new HostnamePortSpecification(uri.getHost()+":"+uri.getPort(), null));
-                }else{
-                    domainSpecifications.add(new HostnameSpecification(uri.getHost(),null));
-                }
-                domainSpecifications.add(new SchemeSpecification(uri.getScheme()));
-
-                boolean result = store.addDomain(new Domain(DOMAIN_NAME,
-                        "Github Domain to store personal access token",
-                        domainSpecifications
-                ));
-                if(!result){
-                    throw new ServiceException.BadRequestExpception("Github accessToken is valid but no valid credential domain found and could not be created");
-                }
-                domain = store.getDomainByName(DOMAIN_NAME);
-                if(domain == null){
-                    throw new ServiceException.BadRequestExpception("Github accessToken is valid but no valid credential domain found and could not be created");
-                }
-            }
-
-            if(githubCredential == null){
-                if(!store.addCredentials(domain, credential)){
-                    throw new ServiceException.UnexpectedErrorException("Failed to add credential to domain");
-                }
-
+            if(githubCredential == null) {
+                CredentialsUtils.createCredentialsInUserStore(
+                        credential, authenticatedUser, DOMAIN_NAME, new URI(getUri()));
```

Comment:
Will the `getUri()` return the GitHub Enterprise URL if the user is creating a pipeline against GHE?